cmd/thor/node/packer_loop.go

```diff
@@ -211,3 +219,24 @@ func (n *Node) pack(flow *packer.Flow) (err error) {
 		return newBlock.Header().ID(), nil
 	})
 }
+
+func (n *Node) completeSync() error {
+	writer := n.logDB.NewWriter()
+	logger.Info("Sync done, creating indexes")
+	err := writer.ExecuteJournalWalModeSwitch()
```

Comment:
Ok, maybe that is why this line exists? If that is the case, maybe it makes sense to just remove the WAL assignment at `New`.